<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" colspan="3">
     <layout class="QHBoxLayout" name="horizontalLayout_7">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QPushButton" name="add_node">
            <property name="text">
             <string>Add Node</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="to_tree">
            <property name="text">
             <string>To Tree</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Data</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="data"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Key </string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="key"/>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>37</width>
          <height>17</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="delete_node">
            <property name="text">
             <string>Delete Node</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="del"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPushButton" name="find_node">
            <property name="text">
             <string>Find Node</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="find"/>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>16</width>
          <height>17</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="pint_tree">
        <property name="text">
         <string>Print Tree</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QRadioButton" name="pre_order">
            <property name="text">
             <string>Pre-oreder</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="in_order">
            <property name="text">
             <string>In-oreder</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="post_order">
            <property name="text">
             <string>Post-order</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="balance_tree">
      <property name="text">
       <string>Balance Tree</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="fill_from_table">
      <property name="text">
       <string>Feel From Table</string>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QPushButton" name="del_tree">
      <property name="text">
       <string>Delete Tree</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0" colspan="3">
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QTableWidget" name="tableWidget">
            <column>
             <property name="text">
              <string>Data</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Key</string>
             </property>
            </column>
           </widget>
          </item>
          <item>
           <widget class="QTreeWidget" name="treeView">
            <property name="columnCount">
             <number>1</number>
            </property>
            <column>
             <property name="text">
              <string>                                         Your Tree                                                                     </string>
             </property>
            </column>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="textBrowser"/>
      </item>
      <item>
       <widget class="QPushButton" name="solve_task">
        <property name="text">
         <string>Solve Task</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
